multihost/db-postgresql-with-custom-certificates/test.sh

```diff
@@ -0,0 +1,375 @@
+    keylime_server_registrar_database_url: \"postgresql://registrar:regi@${ATTESTATION_SERVER_IP}/registrardb\"
+
+  roles:
+    - rhel-system-roles.keylime_server
+EOF"
+
+        rlRun 'ansible-playbook -v --ssh-common-args "-o StrictHostKeychecking=no" -i inventory playbook.yml'
+        rlRun "sync-set ANSIBLE_SETUP_DONE"
+    rlPhaseEnd
+
+}
+
+Agent() {
+    rlPhaseStartSetup "Agent setup"
+        AGENT_ID="d432fbb3-d2f1-4a97-9ef7-75bd81c00000"
+        CV_CA=/var/lib/keylime/cv_ca
+        rlRun "mkdir -p $CV_CA"
+        rlRun "sync-block ATTESTATION_SERVER_START ${ATTESTATION_SERVER_IP}" 0 "Waiting for the attestation server finish to start"
+        rlRun "cp /var/lib/keylime/cacert.pem /var/lib/keylime/cv_ca/"
+        id keylime && rlRun "chown -R keylime:keylime ${CV_CA}"
+
+        rlRun "limeUpdateConf agent ip '\"${AGENT_IP}\"'"
+        rlRun "limeUpdateConf agent contact_ip '\"${AGENT_IP}\"'"
+        rlRun "limeUpdateConf agent registrar_ip '\"${ATTESTATION_SERVER_IP}\"'"
+        rlRun "limeUpdateConf agent trusted_client_ca '\"/var/lib/keylime/cv_ca/cacert.pem\"'"
+        rlRun "limeUpdateConf agent revocation_notification_ip '\"${ATTESTATION_SERVER_IP}\"'"
+
+        if [ -n "$KEYLIME_TEST_DISABLE_REVOCATION" ]; then
+            rlRun "limeUpdateConf agent enable_revocation_notifications False"
+        fi
+
+        # Delete other components configuration files
+        for comp in verifier registrar tenant; do
+            rlRun "rm -rf /etc/keylime/$comp.conf*"
+        done
+
+        # if TPM emulator is present
+        if limeTPMEmulated; then
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            rlRun "limeCondStartAbrmd"
+            # start ima emulator
+            limeInstallIMAConfig
+            rlRun "limeStartIMAEmulator"
+        fi
+        sleep 5
+        limeCreateTestPolicy
+        #copy policy.json to tenant for keylime tenant add
+        rlRun "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null policy.json root@$ATTESTATION_SERVER_IP:/var/tmp/"
+        rlRun "limeStartAgent"
+        rlRun "sync-set AGENT_STARTS"
+    rlPhaseEnd
+
+    rlPhaseStartTest "keylime attestation test: Check if agent is added"
+        rlRun "sync-block AGENT_ADD_DONE ${ATTESTATION_SERVER_IP}" 0 "Waiting for adding agent."
+        rlWaitForFile /var/tmp/test_payload_file -t 30 -d 1  # we may need to wait for it to appear a bit
+        rlAssertExists /var/tmp/test_payload_file
+    rlPhaseEnd
+
+
+    rlPhaseStartTest "Agent attestation test: Fail keylime agent"
+        # fail AGENT and confirm it has failed validation
+        TESTDIR=`limeCreateTestDir`
+        limeExtendNextExcludelist $TESTDIR
+        rlRun "echo -e '#!/bin/bash\necho boom' > $TESTDIR/keylime-bad-script.sh && chmod a+x $TESTDIR/keylime-bad-script.sh"
+        rlRun "$TESTDIR/keylime-bad-script.sh"
```

Comment:
```
rlRun "sync-set AGENT_CORRUPTED"
rlRun "sync-block AGENT_FAILED ${ATTESTATION_SERVER_IP}"

Reply:
And where you want set "`syn-block AGENT_CORRUPTED ${ATTESTATION_SERVER_IP}`"?

Reply:
Check line 179 above.

Reply:
Above checking if agent failed on verifier side?